frontend/src/app/sessions/session/tiling-window-manager/tiling-window-manager.component.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, HostListener, Input, OnInit } from '@angular/core';
+import { Session } from 'src/app/schemes';
+import { FullscreenService } from '../../service/fullscreen.service';
+
+@Component({
+  selector: 'app-tiling-window-manager',
+  templateUrl: './tiling-window-manager.component.html',
+  styleUrls: ['./tiling-window-manager.component.css'],
+})
+export class TilingWindowManagerComponent implements OnInit {
+  private _tilingSessions: TilingSession[] = [];
+
+  @Input()
+  set sessions(sessions: Session[]) {
+    this._tilingSessions = sessions.map((session, index) => ({
+      ...session,
+      index: index,
+      width: -1,
+      fullscreen: false,
+    }));
+    this.resetWidths();
+  }
+
+  get sessions(): TilingSession[] {
+    return this._tilingSessions.sort((a, b) => a.index - b.index);
+  }
+
+  public resizeState: ResizeState = {};
+
+  private minimumSessionWidth = 0;
+  private existingMargin = 0;
+
+  constructor(public fullscreenService: FullscreenService) {}
+
+  ngOnInit(): void {
+    this.fullscreenService.isFullscreen$.subscribe((isFullscreen) => {
+      this.existingMargin = isFullscreen ? 0 : 27.48;
+      this.resetWidths();
+    });
+  }
+
+  onMouseDown(event: MouseEvent, index: number): void {
+    const leftSession = this.getSessionByIndex(index);
+    const rightSession = this.getSessionByIndex(index + 1);
+
+    if (leftSession && rightSession) {
+      this.resizeState = {
+        index: index,
+        startX: event.clientX,
+        leftSession: leftSession,
+        rightSession: rightSession,
+        startWidthLeft: leftSession.width,
+        startWidthRight: rightSession.width,
+      };
+      this.toggleIFrameEvents(false);
+    }
+  }
+
+  @HostListener('window:mousemove', ['$event'])
+  onMouseMove(event: MouseEvent): void {
+    if (this.isValidResizeState(this.resizeState)) {
+      const delta = event.clientX - this.resizeState.startX;
+      const [newWidthLeft, newWidthRight] = this.calculateNewWidths(
+        this.resizeState,
+        delta,
+      );
+
+      this.resizeState.leftSession.width = newWidthLeft;
+      this.resizeState.rightSession.width = newWidthRight;
+    }
+  }
+
+  @HostListener('window:mouseup')
+  onMouseUp(): void {
+    this.resizeState = {};
+    this.toggleIFrameEvents(true);
+  }
+
+  @HostListener('window:resize')
+  onResize() {
+    this.resetWidths();
+  }
+
+  onLeftClick(session: TilingSession) {
```

Comment:
We should use another name for it. "LeftClick" is for me the "left" mouse button and "RightClick" the right mouse button.

Reply:
Good point! I renamed it to `onLeftArrowClick` and `onRightArrowClick` which should be more clear in this context.